### PR TITLE
Cache control

### PR DESCRIPTION
**Pull request by pvalsecc**

### CODE REVIEW

File: core/src/main/java/org/mapfish/print/servlet/BaseMapServlet.java [hunk 1]
```diff
@@ -117,4 +118,28 @@ protected final StringBuilder getBaseUrl(final HttpServletRequest httpServletReq
...
+    public final void setCacheDuration(final int cacheDurationInSeconds) {
+        this.cacheDurationInSeconds = cacheDurationInSeconds;
+    }
+
+    /**
+     * Disable caching of the response.
+     * @param response the response
+     */
+    protected void setNoCache(final HttpServletResponse response) {
+        response.setHeader("Cache-Control", "no-cache");
```
Comment (mfournier) [inline core/src/main/java/org/mapfish/print/servlet/BaseMapServlet.java]: NB: `no-cache` is a value usually found in requests, not responses. I'm not sure of the effect in this case.

To be on the safe side, I'd update this to read `private, max-age=0` instead.

Comment (pvalsecc) [inline core/src/main/java/org/mapfish/print/servlet/BaseMapServlet.java]: When I ask Pyramid (python framework we use)  to not allow the caching of its answer it adds that:
```
Cache-Control: max-age=0, must-revalidate, no-cache, no-store
```

And the RFC 7234 says about `private`:
```
A private cache MAY store the response and reuse it for later requests, even if the response
would normally be non-cacheable.
```

We don't want that for our polling URLs, at least.

But OK, Ill change the PR to go halfway in your direction and use the same header as Pyramid.

File: examples/src/test/resources/examples/datasource_multiple_maps/expected_output/image-similarity.txt [hunk 2]
```diff
@@ -1 +1 @@
-175
+190
```
Comment (Vampouille) [inline examples/src/test/resources/examples/datasource_multiple_maps/expected_output/image-similarity.txt]: Why did you change image similarity, does the cache affects generated images ?

Comment (pvalsecc) [inline examples/src/test/resources/examples/datasource_multiple_maps/expected_output/image-similarity.txt]: Not at all, I just need that to run the tests on my machine.

Comment (Vampouille) [inline examples/src/test/resources/examples/datasource_multiple_maps/expected_output/image-similarity.txt]: So image generation depends on host used ? Different hosts generates different images ?

Comment (pvalsecc) [inline examples/src/test/resources/examples/datasource_multiple_maps/expected_output/image-similarity.txt]: I'm afraid so. Maybe the fonts, maybe the java version, maybe the OS. We used to have different expected images for windows. Now we don't care about windows builds...

Comment (pvalsecc) [inline examples/src/test/resources/examples/datasource_multiple_maps/expected_output/image-similarity.txt]: I've compared visually the actual and the expected and I see no difference.
